### PR TITLE
Workaround intel/17 internal compiler error with tool utils

### PR DESCRIPTION
**Pull request by ndellingwood**
Addresses issue #4612

### CODE REVIEW

File: core/unit_test/tools/include/ToolTestingUtilities.hpp [hunk 1]
```diff
@@ -267,9 +267,19 @@ struct invoke_helper {
  */
 inline MatchDiagnostic check_match(event_vector::size_type events_scanned,
                                    const event_vector& events) {
-  return (events_scanned == events.size())
-             ? MatchDiagnostic{true}
-             : MatchDiagnostic{false, {"Wrong number of events encountered"}};
+// Workaround intel/17 internal error
+// See https://github.com/kokkos/kokkos/issues/4612
+#ifndef KOKKOS_COMPILER_INTEL
```
Comment (dalg24) [inline core/unit_test/tools/include/ToolTestingUtilities.hpp:272]:
```suggestion
#if defined(KOKKOS_COMPILER_INTEL) && (KOKKOS_COMPILER_INTEL < 1800)
```

Comment (ndellingwood) [reply core/unit_test/tools/include/ToolTestingUtilities.hpp]: @dalg24 thanks for feedback, force pushed the update 54112f4fa8b1073f8a503f1ffa9e963dd8fc4711

**Review by crtrott — APPROVED** [review]

File: core/unit_test/tools/include/ToolTestingUtilities.hpp [hunk 2]
```diff
@@ -267,9 +267,19 @@ struct invoke_helper {
...
+  auto result =
+      ((events_scanned == events.size())
+           ? MatchDiagnostic{true}
+           : MatchDiagnostic{false, {"Wrong number of events encountered"}});
+  return result;
+#else
+  return ((events_scanned == events.size())
+              ? MatchDiagnostic{true}
+              : MatchDiagnostic{false, {"Wrong number of events encountered"}});
+#endif
```
Comment (masterleinad) [inline core/unit_test/tools/include/ToolTestingUtilities.hpp:282]: Why not just
```suggestion
  auto result =
      ((events_scanned == events.size())
           ? MatchDiagnostic{true}
           : MatchDiagnostic{false, {"Wrong number of events encountered"}});
  return result;
```
?

Comment (ndellingwood) [reply core/unit_test/tools/include/ToolTestingUtilities.hpp:282]: That's fine too :)

Comment (ndellingwood) [reply core/unit_test/tools/include/ToolTestingUtilities.hpp:282]: I force pushed the cleaner update 08549b84299a2fc2e6041527ca4db72926d7ffd7

**Review by masterleinad — APPROVED** [review]

**Comment by crtrott** [comment]
Retest this please.

**Pull request merged**